qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -555,22 +555,17 @@ def _is_target_series(datum, **filters):
                 f"X value key {x_key} is not defined in circuit metadata."
             ) from ex
 
-        y_values, y_sigmas = zip(*map(data_processor, data))
-
-        # TODO this should be handled in data processor.
-        # Future data processor may take full sequence of data rather than datum.
-        # The CurveAnalysis can pass series filter_kwargs to the processor
-        # so that it can filter data to extract.
-        if self._get_option("normalization"):
-            y_min, y_max = min(y_values), max(y_values)
-            scale = 1 / (y_max - y_min)
+        if isinstance(data_processor, DataProcessor) and data_processor.requires_all_data():
```

Comment:
I think this still requires filter keys. In the current data processor model, we can perform post processing (for example normalization) on filtered series data. On the other hand, this cannot perform such processing without filter key. Do you think such series-wise processing is not necessary?